### PR TITLE
New extract() function to simplify executeInWorkerPool() results hand…

### PR DESCRIPTION
**Pull request by jlrobins**
…ling when expecting all to have succeeded.

## Summary of Changes

- `extract()` results post-processor giving `executeInWorkerPool()` similar semantics to `Promise.all()` if the use-case desires: It unpacks the actual results from the `ExecutionResult<R>` wrappers into an `R[]`. If an `ErrorResult` is encountered, it immediately throws it.
- Use to replace equivalent code inlined within the tail of `fetchSchemaSubjectGroup()`. Existing tests over it continue to prove that if any single version fetch failed, the whole block fails.
- While in here, fix edge case in executeInWorkerPool(), which would leave result array values undefined if the callback raised a non-Error. Fix that by noticing and wrapping with a definite Error.

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

##### Tests

- [x] Added new
- [ ] Updated existing
- [ ] Deleted existing

##### Other

- [ ] All new disposables (event listeners, views, channels, etc.) collected as  for eventual cleanup?

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md) or [README](https://github.com/confluentinc/vscode/blob/main/public/README.md)?
- [ ] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/README.md#packaging-steps) and installing the extension `.vsix` file?
  ```shell
  gulp clicktest
  ```

### CODE REVIEW

File: src/utils/workerPool.ts [hunk 1]
```diff
@@ -145,3 +145,24 @@ export async function executeInWorkerPool<T, R>(
...
+ *
+ * Makes {@link executeInWorkerPool} error handling more like `Promise.all()` when the caller
+ * desires all-or-none semantics.
+ */
+export function extract<R>(results: ExecutionResult<R>[]): R[] {
+  const goodResults: R[] = new Array<R>(results.length);
+  let i = 0;
+  for (const result of results) {
+    if (isSuccessResult(result)) {
+      goodResults[i++] = result.result;
```
Comment (shouples) [inline src/utils/workerPool.ts:161]: If `results` are returned in order, do we need to insert at specific index values rather than just `.push()`ing?

Comment (jlrobins) [reply src/utils/workerPool.ts:161]: avoiding `push()` and preallocating the properly sized array means no array expansion / reallocation.

File: src/utils/workerPool.ts [hunk 2]
```diff
@@ -145,3 +145,24 @@ export async function executeInWorkerPool<T, R>(
...
+    } else {
+      throw result.error;
+    }
```
Comment (shouples) [inline src/utils/workerPool.ts:164]: May want to explicitly check for `isErrorResult` here since it's possible to get an `undefined` result back from `executeInWorkerPool()` if an `undefined` was passed into the items array.

Test with that behavior: https://github.com/confluentinc/vscode/blob/213d4731d255485e834928ebb6fc2e99832efc60/src/utils/workerPool.test.ts#L113-L121

**Review by shouples — APPROVED** [review]
